psc-ide.el

```diff
@@ -782,9 +782,12 @@ on whether WARN is true.  Optionally EXPANDs type synonyms."
 
 (defun psc-ide-suggest-project-dir ()
   "Suggest a project directory to start the ide server in."
-  (if (and (fboundp 'projectile-project-root) (projectile-project-p))
-      (projectile-project-root)
-    default-directory))
+  (cond
+   ((and (fboundp 'projectile-project-root) (projectile-project-p))
+    (projectile-project-root))
+   ((fboundp 'project-root)
+     (project-root (project-current)))
+   (t default-directory)))
```

Comment:
`project-root` is new in `project` version `0.3.0`, which can be installed from [GNU ELPA](http://elpa.gnu.org/packages/project.html) in Emacs 26+, but won't be built-in before Emacs 28.

Furthermore, I don't think these `fboundp` checks are sufficient, because neither `project-root` nor `projectile-project-p` (which is just a thin wrapper around `projectile-project-root`) is autoloaded.

Finally, shouldn't the built-in `project` be checked before the third-party `projectile`?

Maybe something like the following 100% guaranteed untested code:

```el
(or (let ((pr (project-current)))
      (cond ((not pr) nil)
            ;; New in project.el v0.3.0.
            ((fboundp 'project-root) (project-root pr))
            ((with-no-warnings (car (project-roots pr))))))
    (and (require 'projectile nil t)
         (fboundp 'projectile-project-root)
         (projectile-project-root))
    default-directory)
```